src/lib/mo-pdf-viewer.component.ts

```diff
@@ -1,102 +1,143 @@
+    NgxExtendedPdfViewerModule,
+    NgbModule,
+    FormsModule,
   ],
+  providers: [provideAnimations(), UtilService],
   selector: 'mo-pdf-viewer',
   templateUrl: './mo-pdf-viewer.component.html',
   styleUrls: ['./mo-pdf-viewer.component.scss'],
 })
-export class MoPdfViewerComponent {
+export class MoPdfViewerComponent implements OnDestroy, OnInit {
   @Input({ required: true }) public pdfSrc: string | Uint8Array = '';
-  @Input() public documentTitle = '';
-  @Input() public documentClassification = 'Unclassified';
-  @Input() public documentAuthor = '';
-  @Input() public minimalView = false;
-  @ViewChild(PdfViewerComponent) public pdfComponent!: PdfViewerComponent;
-  public currentSearchIndex: number = 1;
-  public totalMatchesCount: number = 0;
+  @Input() public set annotations(savedAnnotations: AnnotationItem[]) {
+    if (savedAnnotations && savedAnnotations.length > 0) {
+      this.savedAnnotations = savedAnnotations;
+      this.notRenderedPages = [
+        ...new Set(this.savedAnnotations.map((a) => a.pageNumber)),
+      ];
+
+      this.requestedPages = [...this.notRenderedPages];
+    }
+  }
+  @Input() public savedAnnotations: AnnotationItem[] = [];
+  @Output() public annotationUpdated = new EventEmitter<AnnotationItem[]>();
+  public tagListPublic: TagListModel[] = [];
+  public tagListPrivate: TagListModel[] = [];
+  public commentList: highlightEditor[] = [];
+  public isOpenTag: boolean = false;
+  public isOpenComment: boolean = false;
+  public publicListVisible: boolean[] = [];
+  public privateListVisible: boolean[] = [];
+  public isHovered: boolean = false;
+  public tags: TagModel[] = [];
+  private notRenderedPages: number[] = [];
+  public subscription: Subscription | undefined;
+  private requestedPages: number[] = [];
+  @ViewChild(NgxExtendedPdfViewerComponent)
+  public pdfComponent!: NgxExtendedPdfViewerComponent;
+
+  @ViewChild(TagPopoverComponent)
+  public tagPopoverComponent!: TagPopoverComponent;
+
+  public ngOnInit(): void {
+    const pdfViewerElement = this.elementRef.nativeElement.querySelector('ngx-extended-pdf-viewer');
+    pdfViewerElement.addEventListener('scroll', this.onPdfViewerScroll);
+  }
+
+  private popoverRef: ComponentRef<
+    CommentPopoverComponent | TagPopoverComponent
+  > | null = null;
+  public previousScrollTop = 0;
+  public previousScrollLeft = 0;
+  public highlightList: highlightEditor[] = [];
+  public hoveredList: 'private' | 'public' | null | 'highlight' = null;
+  public hoveredIndex: number | null = null;
+  private callExecuted: boolean = false;
+
+  constructor(
+    private viewContainerRef: ViewContainerRef,
+    private appRef: ApplicationRef,
+    private elementRef: ElementRef,
+    public utilService: UtilService,
+    public renderer: Renderer2,
+    private pdfService: NgxExtendedPdfViewerService
+  ) {
+    this.privateListVisible = new Array(this.tagListPrivate.length).fill(false);
+
+    this.utilService.annotationUpdated$.subscribe(() => {
```

Comment:
add `.pipe(takeUntilDestroyed())` to these subscriptions to avoid leaks. (from Angular rxjs interlop)